libdnf5/rpm/package_query.cpp

```diff
@@ -2828,4 +2828,52 @@ void PackageQuery::filter_installonly() {
     filter_provides(installonly_packages, libdnf5::sack::QueryCmp::GLOB);
 }
 
+static const std::unordered_set<std::string> CORE_PACKAGE_NAMES = {
+    // See https://access.redhat.com/solutions/27943. These packages should all
+    // also have a reboot_suggested advisory, but it's good to handle them
+    // explicitly, just to be sure.
+    "kernel",
+    "kernel-core",
+    "kernel-PAE",
+    "kernel-rt",
+    "kernel-smp",
+    "kernel-xen",
+    "linux-firmware",
+    "dbus",
+    "glibc",
+    "hal",
+    "systemd",
+    "udev",
+    "gnutls",
+    "openssl-libs",
+    "dbus-broker",
+    "dbus-daemon",
+};
+
+void PackageQuery::filter_reboot_suggested() {
+    auto & pool = get_rpm_pool(p_impl->base);
+    libdnf5::solv::SolvMap filter_result{pool.get_nsolvables()};
+
+    for (const auto & pkg : *this) {
+        if (CORE_PACKAGE_NAMES.contains(pkg.get_name())) {
+            filter_result.add_unsafe(pkg.get_id().id);
+        }
+    }
+
+    libdnf5::advisory::AdvisoryQuery advisories{p_impl->base};
+    auto reboot_advisories =
+        advisories.get_advisory_packages_sorted_by_name_arch_evr() |
+        std::views::filter([](const auto & advisory_pkg) { return advisory_pkg.get_reboot_suggested(); });
```

Comment:
So I was testing the new `needs-restarting` plugin in CLI inside a Fedora container and I experienced a VERY slow execution time when running the `dnf5 needs-restarting`, like above 10 seconds compared to <1s using the `dnf4 needs-restarting -r`. I figured out it is because of using this range view. I didn't analyse what is going on in the background, but just changing this line to `std::erase_if(reboot_advisories, [](const auto & pkg) { return !pkg.get_reboot_suggested(); });` results again in similar execution time as with dnf4.